odoo/addons/test_inherit_depends/tests/test_inherit_depends.py

```diff
@@ -0,0 +1,20 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import common
+
+class test_inherit(common.TransactionCase):
+    def test_ir_model_data_xmlid_two_ways_inheritance_with_mixin(self):
+        """ Check the XMLID exists one the concrete model """
+
+        # Test the situation where there is a mixin X defined in module A,
+        # a model M defined in module B and that a third module C extends M
+        # to depends on C. The XMLID for X on C must exist.
+        # A => test_new_api (just because it is handy to reuse it)
+        # B => test_inherit
+        # C => test_inherit_depends
+        # M => test_new_api.foo
+        # X => test.inherit.mixin
+        IrModelData = self.env['ir.model.data']
+        field = IrModelData.search([('name', '=', 'field_test_new_api_bazz__published')])
+        self.assertEqual(len(field), 1)
+        self.assertEqual(field.module, 'test_inherit_depends')
```

Comment:
This is not a field: `field` → `imd` or `data` or ?
There is an alternative way to retrieve XML ids of a given record:
```suggestion
        field = self.env['ir.model.fields']._get('test_new_api.bazz', 'published')
        self.assertTrue(field)
        self.assertEqual(field._get_external_ids()[field.id],
                         ['test_inherit_depends.field_test_new_api_bazz__published'])
```